### PR TITLE
[fix] Only trigger externalAcctSignup event when a new user is created

### PR DESCRIPTION
**Pull request by pjlast**
Currently events are triggered whenever a user signs in with `http-header` auth. This is because of the `GetAndSaveUser` function always triggering an event.

However, before the new telemetry events, these events were only created when a new user was created.

This PR brings the new telemetry code in line with the old telemetry code to stop the massive amounts of spam caused by this event.

Closes SRC-461

## Test plan

Adjust expected events in unit test.

## Changelog

- Fixed an issue where the `http-header` auth would cause a massive amount of event logs spam

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
I want Robert to review this, but code change in general LGTM, without knowing what the event was supposed to track 😬

**Review by eseliger — COMMENTED** [review]
ah: changelog entry :)

File: cmd/frontend/auth/user.go [hunk 1]
```diff
@@ -260,38 +260,42 @@ func GetAndSaveUser(
...
-					"extAcctSaved": telemetry.Bool(extAcctSaved),
-				},
-				op.UserCreateEventProperties,
-			),
-			PrivateMetadata: privateMetadata,
-		})
-	}()
+			// Record our V2 event.
+			recorder.Record(telemetryCtx, telemetryV2UserSignUpFeatureName, action, &telemetry.EventParameters{
+				Version: 2, // We've significantly refactored telemetryV2UserSignUpFeatureName occurrences
```
Comment (bobheadxi) [inline cmd/frontend/auth/user.go:285]:
```suggestion
				Version: 3, // We've significantly refactored telemetryV2UserSignUpFeatureName occurrences
```

File: cmd/frontend/auth/user.go [hunk 2]
```diff
@@ -260,38 +260,42 @@ func GetAndSaveUser(
...
+				Metadata: telemetry.MergeMetadata(
+					telemetry.EventMetadata{
+						"serviceVariant": telemetry.Number(serviceVariant),
+						// Track the various outcomes of the massive signup closure above.
+						"newUserSaved": telemetry.Bool(newUserSaved),
```
Comment (bobheadxi) [inline cmd/frontend/auth/user.go:290]: Do we still need `newUserSaved` in the metadata, if we only record the event when `newUserSaved`?

**Review by bobheadxi — APPROVED** [review]
Thanks for digging into this! Overall LGTM, but - doesn't this mean that we no longer record telemetry when you sign up with a new external account into an existing user?

i.e. I sign in with GitHub, then sign in with GitLab into the same user, we should now only get one event based on a reading of the code, and there will be no record of the second sign-in with GitLab. Maybe that's what we want anyway?

**Comment by pjlast** [comment]
@bobheadxi we can do `newUserSaved || extAcctAdded` (or whatever the variable names are). I just went with what the legacy telemetry did. But if we want as much telemetry as possible (without overloading the system ofc) then we can go this route instead

**Comment by bobheadxi** [comment]
@pjlast what you suggest sounds good to me! I also don't know how much data we want from this specifically, so I defer to your judgement as to what is useful as owner of this stuff. You could also check in with someone from the data analytics team about the specific scenarios they might be interested in, as the last time I was in this code is because they were looking at one of the events emitted here

**Pull request merged**